fabtests/component/dmabuf-rdma/fi-mr-reg-rocr.c

```diff
@@ -0,0 +1,264 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <string.h>
+#include <rdma/fabric.h>
+#include <rdma/fi_domain.h>
+#include <rdma/fi_endpoint.h>
+#include "util.h"
+#include "rocr.h"
+
+static int	ep_type = FI_EP_RDM;
+static char	*prov_name;
+static char	*domain_name;
+
+static struct fi_info		*fi;
+static struct fid_fabric	*fabric;
+static struct fid_eq		*eq;
+static struct fid_domain	*domain;
+static struct fid_ep		*ep;
+static struct fid_av		*av;
+static struct fid_cq		*cq;
+static struct fid_mr		*mr, *dmabuf_mr;
+
+static void	*buf;
+static size_t	buf_size = 65536;
+static int 	buf_location = MALLOC;
+
+static void init_buf(void)
+{
+	int page_size = sysconf(_SC_PAGESIZE);
+
+	buf = rocr_alloc_buf(page_size, buf_size, buf_location, 0, NULL);
+	if (!buf) {
+		fprintf(stderr, "Couldn't allocate work buf.\n");
+		exit(-1);
+	}
+}
+
+static void free_buf(void)
+{
+	rocr_free_buf(buf, buf_location);
+}
+
+static void init_ofi(void)
+{
+	struct fi_info *hints;
+	struct fi_cq_attr cq_attr = {};
+	struct fi_av_attr av_attr = {};
+	struct fi_eq_attr eq_attr = {};
+	int version;
+
+	EXIT_ON_NULL((hints = fi_allocinfo()));
+
+	hints->caps = FI_HMEM;
+	hints->ep_attr->type = ep_type;
+	if (prov_name)
+		hints->fabric_attr->prov_name = strdup(prov_name);
+	hints->domain_attr->mr_mode = (FI_MR_ALLOCATED | FI_MR_PROV_KEY |
+				       FI_MR_VIRT_ADDR | FI_MR_LOCAL |
+				       FI_MR_HMEM | FI_MR_ENDPOINT);
+	if (domain_name)
+		hints->domain_attr->name = strdup(domain_name);
+
+	version = FI_VERSION(1, 12);
+
+	if (ep_type == FI_EP_RDM)
+		EXIT_ON_ERROR(fi_getinfo(version, NULL, NULL, 0, hints, &fi));
+	else
+		EXIT_ON_ERROR(fi_getinfo(version, "localhost", "12345", 0,
+					 hints, &fi));
+
+	fi_freeinfo(hints);
+
+	printf("Using OFI device: %s (%s)\n", fi->fabric_attr->prov_name,
+		fi->domain_attr->name);
+
+	EXIT_ON_ERROR(fi_fabric(fi->fabric_attr, &fabric, NULL));
+	EXIT_ON_ERROR(fi_domain(fabric, fi, &domain, NULL));
+	EXIT_ON_ERROR(fi_endpoint(domain, fi, &ep, NULL));
+	EXIT_ON_ERROR(fi_cq_open(domain, &cq_attr, &cq, NULL));
+	if (ep_type == FI_EP_RDM) {
+		EXIT_ON_ERROR(fi_av_open(domain, &av_attr, &av, NULL));
+		EXIT_ON_ERROR(fi_ep_bind(ep, (fid_t)av, 0));
+	} else {
+		EXIT_ON_ERROR(fi_eq_open(fabric, &eq_attr, &eq, NULL));
+		EXIT_ON_ERROR(fi_ep_bind(ep, (fid_t)eq, 0));
+	}
+	EXIT_ON_ERROR(fi_ep_bind(ep, (fid_t)cq, (FI_TRANSMIT | FI_RECV |
+						 FI_SELECTIVE_COMPLETION)));
+	EXIT_ON_ERROR(fi_enable(ep));
+}
+
+void reg_mr(void)
+{
+	struct iovec iov = {
+		.iov_base = buf,
+		.iov_len = buf_size,
+	};
+	struct fi_mr_attr mr_attr = {
+		.mr_iov = &iov,
+		.iov_count = 1,
+		.access = FI_REMOTE_READ | FI_REMOTE_WRITE,
+		.requested_key = 1,
+		.iface = buf_location == MALLOC ? FI_HMEM_SYSTEM : FI_HMEM_ROCR,
+		//.device.ze = rocr_get_dev_num(0),
```

Comment:
`.device.rocr`

Reply:
[`fi_mr_aatr` ](https://github.com/ofiwg/libfabric/blob/main/include/rdma/fi_domain.h#L170) dont have a field for rocr